docs/sdk/guides/03-build-your-first-cross-chain-dapp.md

```diff
@@ -2,165 +2,129 @@
-
-1. [Connect your MetaMask to Pangoro](./01-using-smart-chain-with-metamask.md).
-2. Export you Dapp to a single flattened solidity file.
-    
-    ```bash
-    npx hardhat flatten ./contracts/RemarkDemo.sol > ~/Desktop/RemarkDemo.sol
-    ```
-    
-3. Copy it to Remix and deploy the `RemarkDemo` contract.
-4. Run `systemRemark` with value 200 ethers(here means ORINGs), the value must ≥ the [market fee](../../fee.md). important!
-
-## Or, Run in your project with hardhat command
-
-1. Open your `hardhat.config.js` in your project folder, add Crab Network into `networks`.
-
-    ```js
-    crab: {
-      url: 'https://pangoro-rpc.darwinia.network',
-      network_id: "45",
-      accounts: [PRIVATE_KEY],
-      gas: 3_000_000,
-      gasPrice: 53100000000
-    }
-    ```
-    PRIVATE_KEY is your private key of Crab Network account with enough(≥ market fee + transaction fee) ORINGs tokens.
-
-2. Add a script to deploy and run your contract. Inside `scripts/`, create a new file `remarkdemo-script.js` with the following code:
-
-    ```js
-    const hre = require("hardhat");
-
-    async function main() {
-        // We get the contract to deploy
-        const RemarkDemo = await hre.ethers.getContractFactory("RemarkDemo");
-        const demo = await RemarkDemo.deploy();
-        await demo.deployed();
-        await demo.deployTransaction.wait();
-        console.log("Deployed to:", demo.address);
-
-        // Send transaction
-        const tx = await demo.remark({
-          value: BigInt(200000000000000000000), // 200 CRAB, The fee to use the cross-chain service, determined by the Fee Market
-        });
-        await tx.wait();
-        console.log("txhash:", tx["hash"]);
-    }
-
-    main()
-      .then(() => process.exit(0))
-      .catch((error) => {
-        console.error(error);
-        process.exit(1);
-      });
-    ```
-
-3. Run
+Make sure that the latest spec version of pangolin is correct. You can get it from https://pangolin.subscan.io/runtime.
 
-    In your project's folder, run:
+Deploy the Dapp contract on the Pangoro Smart Chain. Inject the endpoint address from previous step into the Dapp contract during its initialization.
 
-    ```bash
-    npx hardhat run --network crab scripts/remarkdemo-script.js
-    ```
+## Run
 
-    It will output the txhash of the transaction. You can copy it to [subscan](https://pangoro.subscan.io/) to see the detail of the transaction. 
+1. You can get a estimated fee by calling the `fee` function of the endpoint contract.
+2. Call the `remoteRemark` with a value. The value should greater than or equal to the estimated fee.
 
 ## Track cross-chain events with subscan
 
 ### MessageAccepted events of Pangoro
 
 [https://pangoro.subscan.io/event?address=&module=bridgedarwiniamessages&event=all](https://crab.subscan.io/event?address=&module=bridgedarwiniamessages&event=all)
 
-### MessageDelivered events of Pangolin
+### MessageDispatched events of Pangolin
```

Comment:
Add more notes about what the events mean in the cross-chain scenarios.

Reply:
There is a doc about the events detail. these is a link to that doc here.